Core/TransactionHandler.php

```diff
@@ -63,16 +64,19 @@ public function __construct($oBackendService)
      *
      * @param Transaction $oParentTransaction
      * @param string      $sActionTitle
-     * @param float       $fAmount
+     * @param float|null  $fAmount
+     * @param array|null  $aOrderItems
      *
      * @return array either [status => success] or [status => error, message => errorMessage]
      *
+     * @throws Exception
+     *
      * @since 1.1.0
      */
-    public function processAction($oParentTransaction, $sActionTitle, $fAmount = null)
+    public function processAction($oParentTransaction, $sActionTitle, $fAmount = null, $aOrderItems = null)
```

Comment:
Way better woad be to have a build transaction, and give this method just the transaction and action. Same way as the frontend transactions are triggered. Its just an advice for the refactoring.

Reply:
that is what `$this->_getPostProcessingTransaction($oParentTransaction, $sActionTitle, $aOrderItems)` line 79 in this method is doing.